### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by tiserett**
[DEMO LINK](https://tiserett.github.io/react_todo-app-loading-todos/)

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,12 +1,55 @@
...
+    }, 3000);
+  }
+
+  let userId = 0;
+
+  if (user?.id) {
+    userId = user?.id;
+  }
+
+  getTodos(userId)
```
Comment (Githi54) [inline src/App.tsx:33]: You can use useEffect, maybe)

**Review by Githi54 — APPROVED** [review]
Well done!🦾

File: src/App.tsx [hunk 2]
```diff
@@ -1,12 +1,55 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
```
Comment (AlinaMatuschak) [inline src/App.tsx:1]: don't disable eslint

File: src/App.tsx (hunk 1 above)
Comment (AlinaMatuschak) [inline src/App.tsx:33]:
```suggestion
  getTodos(user?.id || 0)
```

File: src/App.tsx [hunk 3]
```diff
@@ -1,12 +1,55 @@
...
+
+  let userId = 0;
+
+  if (user?.id) {
+    userId = user?.id;
+  }
+
+  getTodos(userId)
+    .then(setTodos)
+    .catch(() => setError(false));
```
Comment (AlinaMatuschak) [inline src/App.tsx:35]: Let’s use async await. It gives you a simpler way to work with asynchronous promise-based code

File: src/App.tsx [hunk 4]
```diff
@@ -1,12 +1,55 @@
...
+      case FilterType.Completed:
+        return todo.completed;
+
+      default:
+        return null;
+    }
+  });
+
   useEffect(() => {
     // focus the element with `ref={newTodoField}`
```
Comment (AlinaMatuschak) [inline src/App.tsx:54]: delete comment

**Review by AlinaMatuschak — CHANGES_REQUESTED** [review]
Do you just move the code from the last task? There should be just code for the first task )
If you move code, why didn’t you fix comments that I left for your code on the last task?
And check demo link

**Comment by tiserett** [comment]
> Do you just move the code from the last task? There should be just code for the first task ) If you move code, why didn’t you fix comments that I left for your code on the last task? And check demo link

what is the actual reason for changing everything I already changed in the other tasks?

File: src/App.tsx (hunk 2 above)
Comment (AlinaMatuschak) [reply src/App.tsx:1]: why didn’t you fix the comment?

File: src/App.tsx [hunk 5]
```diff
@@ -1,14 +1,52 @@
...
 import { AuthContext } from './components/Auth/AuthContext';
+import { ErrorNotification } from './components/ErrorNotification';
+import { Footer } from './components/Footer';
+import { TodosList } from './components/TodosList';
+import { FilterType } from './types/FilterStatus';
+import { Todo } from './types/Todo';
 
 export const App: React.FC = () => {
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const [error, setError] = useState<boolean | null>(null);
```
Comment (AlinaMatuschak) [inline src/App.tsx:14]:
```suggestion
  const [hasError, setHasError] = useState<boolean>(false);
```

File: src/App.tsx [hunk 6]
```diff
@@ -1,12 +1,55 @@
...
+
+  if (user?.id) {
+    userId = user?.id;
+  }
+
+  getTodos(userId)
+    .then(setTodos)
+    .catch(() => setError(false));
+
+  const filteredTodos = todos.filter(todo => {
```
[the remark this answers is not in the document]
Comment (AlinaMatuschak) [reply src/App.tsx:21]: fix comment

**Review by AlinaMatuschak — APPROVED** [review]
be more attentive to the comments

**Pull request closed without merging**